tests/pytest/test_logic.py

```diff
@@ -0,0 +1,225 @@
+    with pytest.raises(ValueError):
+        int(Logic('Z'))
+
+
+def test_bit_int_conversions():
+    assert int(Bit('0')) == 0
+    assert int(Bit('1')) == 1
+
+
+def test_logic_repr():
+    assert eval(repr(Logic('0'))) == Logic('0')
+    assert eval(repr(Logic('1'))) == Logic('1')
+    assert eval(repr(Logic('X'))) == Logic('X')
+    assert eval(repr(Logic('Z'))) == Logic('Z')
+
+
+def test_bit_repr():
+    assert eval(repr(Bit('0'))) == Bit('0')
+    assert eval(repr(Bit('1'))) == Bit('1')
+
+
+def test_logic_and():
+    # will not be exhaustive
+    assert Logic('0') & Logic('Z') == Logic(0)
+    assert Logic(1) & Logic('1') == Logic(1)
+    assert Logic('X') & Logic('Z') == Logic('X')
+    with pytest.raises(TypeError):
+        Logic('1') & 8
+    with pytest.raises(TypeError):
+        8 & Logic('1')
+
+
+def test_bit_and():
+    assert Bit('0') & Bit('1') == Bit(0)
+    assert Bit(1) & Bit('1') == Bit(1)
+    with pytest.raises(TypeError):
+        Bit('1') & 8
+    with pytest.raises(TypeError):
+        8 & Bit('1')
+
+
+def test_logic_bit_and():
+    r = Logic(0) & Bit(1)
+    assert type(r) == Logic
+    assert r == Logic(0)
+    r = Bit(1) & Logic(0)
+    assert type(r) == Logic
+    assert r == Logic(0)
+
+
+def test_logic_or():
+    # will not be exhaustive
+    assert Logic('1') | Logic('Z') == Logic('1')
+    assert Logic(0) | Logic('0') == Logic(0)
+    assert Logic('X') | Logic('Z') == Logic('X')
+    with pytest.raises(TypeError):
+        8 | Logic(0)
+    with pytest.raises(TypeError):
+        Logic(0) | 8
+
+
+def test_bit_or():
+    assert Bit('0') | Bit('1') == Bit(1)
+    assert Bit(0) | Bit(False) == Bit(0)
+    with pytest.raises(TypeError):
+        8 | Bit(0)
+    with pytest.raises(TypeError):
+        Bit(0) | 8
+
+
+def test_logic_bit_or():
+    r = Logic(0) | Bit(1)
+    assert type(r) == Logic
+    assert r == Logic(1)
+    r = Bit(1) | Logic(0)
+    assert type(r) == Logic
+    assert r == Logic(1)
+
+
+def test_logic_xor():
+    # will not be exhaustive
+    assert (Logic('1') ^ Logic(True)) == Logic(0)
+    assert (Logic(1) ^ Logic('X')) == Logic('X')
+    assert (Logic(1) ^ Logic(False)) == Logic(1)
+    with pytest.raises(TypeError):
+        Logic(1) ^ ()
+    with pytest.raises(TypeError):
+        () ^ Logic(1)
+
+
+def test_bit_xor():
+    assert Bit(0) ^ Bit('1') == Bit(1)
+    assert Bit(False) ^ Bit(0) == Bit('0')
+    with pytest.raises(TypeError):
+        Bit(1) ^ ()
+    with pytest.raises(TypeError):
+        () ^ Bit(1)
+
+
+def test_logic_bit_xor():
+    r = Logic(0) ^ Bit(1)
+    assert type(r) == Logic
+    assert r == Logic(1)
+    r = Bit(0) ^ Logic(0)
+    assert type(r) == Logic
+    assert r == Logic(0)
+
+
+def test_logic_invert():
+    assert ~Logic(0) == Logic(1)
+    assert ~Logic(1) == Logic(0)
+    assert ~Logic('X') == Logic('X')
+    assert ~Logic('Z') == Logic('X')
+
+
+def test_bit_invert():
+    assert ~Bit(0) == Bit(1)
+    assert ~Bit(1) == Bit(0)
```

Comment:
Missing a test for hashability (see if we can stick it in a set). This could also be a good way to test that all the different literals for a value result in the same value by checking if a set holding the values is of size 1 after adding all the literals.